lib/controllers/users.js

```diff
@@ -366,6 +364,65 @@ users.put("/", authenticate, function (req, res, next) {
     });
 });
 
+users.delete("/revoke-device", authenticate, function (req, res, next) {
+    const revokeDeviceOnNMS = (token, deviceType) => {
+      makeRequest({
+        url: `${config.authServiceAPI}/auth/login`,
+        body: {
+          username: config.pushNotificationsServiceUserUsername,
+          password: config.pushNotificationsServiceUserPassword
+        },
+        json: true,
+        headers: {
+          "Content-Type" : "application/json"
+        }
+      }, function(err, httpResponse){
+        if (err) {
+          return next(err);
+        } else if (httpResponse.statusCode !== 200) {
+          res.status(httpResponse.statusCode);
+          return next(errors.UNKNOWN_ERROR);
+        }
+
+        makeRequest({
+          url: `${config.notificationServiceAPI}/users/revoke-device`,
+          form: {
+            username: req.user.email,
+            token,
+            deviceType
+          },
+          headers: {
+            Authorization: "Bearer " + httpResponse.body.token,
+            "Content-Type": "application/json"
+          },
+          json: true
+        }, function(err, httpResponse){
+          if (err) {
+            return next(err);
+          } else if (httpResponse.statusCode !== 200) {
+            res.status(httpResponse.statusCode);
+            return next(errors.UNKNOWN_ERROR);
+          }
+          res.status(200);
+          res.send({
+              success: true
+          });
+        }, "delete");
+      });
+    };
+
+
+    const deviceToken = req.query.deviceToken;
+    const gcmToken = req.query.gcmToken;
+    if (typeof deviceToken !== "undefined" && config.pushNotificationsEnabled) {
+      revokeDeviceOnNMS(deviceToken, "iOS");
+    }
+
+    if (typeof gcmToken !== "undefined" && config.pushNotificationsEnabled) {
+      revokeDeviceOnNMS(gcmToken, "Android");
+    }
```

Comment:
found a bug while functional testing: if neither of the above conditions are true, the request times out. So we should combine these and add an `else` for the `BAD_REQUEST` case. a la

```javascript
    if (typeof deviceToken !== "undefined" && config.pushNotificationsEnabled) {
      revokeDeviceOnNMS(deviceToken, "iOS");
    } else if (typeof gcmToken !== "undefined" && config.pushNotificationsEnabled) {
      revokeDeviceOnNMS(gcmToken, "Android");
    } else {
      next(errors.INVALID_SUBMISSION);
    }
```